src/util/test.cpp

```diff
@@ -335,24 +335,32 @@ CKey AddTestCKeyToKeyStore(CBasicKeyStore& keyStore) {
 }
 
 SpendDescription RandomInvalidSpendDescription() {
-    SpendDescription sdesc;
-    zcash_test_harness_random_jubjub_point(sdesc.cv.begin());
-    zcash_test_harness_random_jubjub_base(sdesc.anchor.begin());
-    sdesc.nullifier = GetRandHash();
-    zcash_test_harness_random_jubjub_point(sdesc.rk.begin());
-    GetRandBytes(sdesc.zkproof.begin(), sdesc.zkproof.size());
-    return sdesc;
+    uint256 cv;
+    uint256 anchor;
+    uint256 rk;
+    libzcash::GrothProof zkproof;
+    SpendDescription::spend_auth_sig_t spendAuthSig;
+    zcash_test_harness_random_jubjub_point(cv.begin());
+    zcash_test_harness_random_jubjub_base(anchor.begin());
+    zcash_test_harness_random_jubjub_point(rk.begin());
+    GetRandBytes(zkproof.begin(), zkproof.size());
+    return SpendDescription(cv, anchor, GetRandHash(), rk, zkproof, spendAuthSig);
```

Comment:
I would use `uint256 nullifier = GetRandHash()` and pass `nullifier` here, just so that it's obvious what it is.

Reply:
This will also be going away in a subsequent PR.